### PR TITLE
rnp: init at 0.14.0-gffcfb63

### PR DESCRIPTION
**Pull request by ribose-jeffreylau**
###### Motivation for this change

Package [rnp](https://github.com/rnpgp/rnp), a high performance C++ OpenPGP library, fully compliant to RFC 4880.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/rnp/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{ lib
+, stdenv
+, botan2
+, bzip2
+, cmake
+, fetchurl
```
Comment (humancalico) [inline pkgs/tools/security/rnp/default.nix:6]:
```suggestion
, fetchFromGitHub
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114238` at 95007f71 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/714dc50b414d51811da275dd5f276aeb">rnp</a></li></li>
  </ul>
</details>

File: pkgs/tools/security/rnp/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+, json_c
+, unzip
+, zlib
+}:
+let
+  version = "0.14.0";
+  name = "rnp-${version}";
+in
+stdenv.mkDerivation rec {
+  inherit name;
```
Comment (humancalico) [inline pkgs/tools/security/rnp/default.nix:16]: You probably don't need to inherit here

File: pkgs/tools/security/rnp/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/rnpgp/rnp/archive/v${version}.zip";
+    sha256 = "0smlr8h13mrn21zhwab49idlb5xz6b6qwsr84asnz2lpx754sfcr";
+  };
```
Comment (humancalico) [inline pkgs/tools/security/rnp/default.nix:21]:
```suggestion
  src = fetchFromGitHub {
    owner = "rnpgp";
    repo = "rnp";
    rev = "v${version}"; 
    sha256 = "0smlr8h13mrn21zhwab49idlb5xz6b6qwsr84asnz2lpx754sfcr"; 
  };
```

File: pkgs/tools/security/rnp/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+, cmake
+, fetchurl
+, json_c
+, unzip
+, zlib
+}:
+let
+  version = "0.14.0";
+  name = "rnp-${version}";
+in
```
Comment (humancalico) [inline pkgs/tools/security/rnp/default.nix:14]: Can you move the `pname` and `version` inside `stdenv.mkDerivation` instead of using a `let ... in`

**Review by akshatagarwl — CHANGES_REQUESTED** [review]
I'm not a committer but still left some reviews.

File: pkgs/tools/security/rnp/default.nix [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+let
+  versionCmake = fetchurl {
+    url = "https://raw.githubusercontent.com/riboseinc/cmake-versioning/c78a0be/version.cmake";
+    sha256 = "09ghbxyf6kr75857a3v9zql5j1zxnw1fkbndwsyrc1k9qmg24824";
+  };
+  rev = "ffcfb6319ffd5dfe0146bc119a3c8eb994c132a2";
+in
+stdenv.mkDerivation rec {
+  pname = "rnp";
+  version = "v0.14.0-1-g${builtins.substring 0 7 rev}+${toString SOURCE_DATE_EPOCH}";
```
Comment (SuperSandro2000) [inline pkgs/tools/security/rnp/default.nix:20]: Please do not prefix the version with v and adding 1970 to the version is also not necessary.

File: pkgs/tools/security/rnp/default.nix [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+  # For generating the correct timestamp in cmake
+  SOURCE_DATE_EPOCH = 1613923200;
```
Comment (SuperSandro2000) [inline pkgs/tools/security/rnp/default.nix:23]: Time stamps are usually not corrected in nixos. Why is this necessary here? I would prefer to just keep it at its default.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/rnp/default.nix (hunk 6 above)
Comment (ribose-jeffreylau) [reply pkgs/tools/security/rnp/default.nix:23]: Thanks for the feedback.

I'd like to know if there's a way to have two kinds of "version" attributes, one for Nix (to keep to nixpkgs's standards) and one for the build process to use (to make it not fail).  If not, then I'll have to come up with something else.

The reason I ask is that since this package is built from a Git commit, according to Nix's packaging guidelines the version should contain `unstable` and the date in `yyyy-mm-dd` format.  However, setting this would break the build process --- it parses its full name and tries to match it with a specific regex pattern, which is unfulfilled by e.g. `unstable-2021-02-21`, but is fulfilled by the string in the above snippet.

Your help is appreciated.

File: pkgs/tools/security/rnp/default.nix (hunk 5 above)
Comment (ribose-jeffreylau) [reply pkgs/tools/security/rnp/default.nix:20]: Please see below.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114238` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>rnp</li>
</ul>
</details>

File: pkgs/tools/security/rnp/default.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/tools/security/rnp/default.nix:23]: You can just patch the right version number in postPatch and give it some dummy version to be happy or widen the regex.

**Comment by ribose-jeffreylau** [comment]
Superceded by 0.15.0

**Comment by SuperSandro2000** [comment]
You could have updated that PR instead.

**Comment by ribose-jeffreylau** [comment]
🙇‍♂️ Sorry, will do next time!
